bin/docker-run

```diff
@@ -170,18 +170,26 @@ fi
 
 # If Docker Host is running ssh-agent, import SSH_AUTH_SOCK into container env
 if env | grep -q SSH_AUTH_SOCK ; then
-  SOCK_PATH="$SSH_AUTH_SOCK" # Default to passing through SSH_AUTH_SOCK (if on angus00.*)
-  if [[ -h $SOCK_PATH ]]; then
-    SOCK_PATH=$(readlink -f $SOCK_PATH)
-  fi
-  # If docker-machine was detected, pass that socket through instead
-  env | grep -q DM_AGENT_SOCK && SOCK_PATH="$DM_AGENT_SOCK"
-  if [[ -z "$SOCK_PATH" || ( "${SOCK_PATH}" != "${DM_AGENT_SOCK}" && ! -e "$SOCK_PATH" ) ]]; then
-     echo "ERROR: SSH_AGENT_SOCK path: $SOCK_PATH does not exist... is your ssh-agent running?"
-  else
-    DOCKER_RUN_SSH_AGENT_ARGS="-v $(dirname $SOCK_PATH):/ssh-agent -e SSH_AUTH_SOCK=/ssh-agent/$(basename $SOCK_PATH)"
+  if uname -s | grep -qi 'darwin' && [ -z "$DOCKER_MACHINE_NAME" ] ; then
+    # Use "magic" ssh-auth.sock path provided as of Docker for Mac >= 2.2.0.0
+    SSH_AUTH_SOCK_MAGIC_PATH='/run/host-services/ssh-auth.sock'
```

Comment:
Notes:
 - Using new "magic" `ssh-auth.sock` path! 🍾 🎆 🎉 👏 
 - Fixed in:
   - docker/for-mac#483
   - docker/for-mac#410